manager/allocator/cnmallocator/networkallocator.go

```diff
@@ -139,16 +139,39 @@ func (na *cnmNetworkAllocator) Allocate(n *api.Network) error {
 	}
 
 	// No swarm-level allocation can be provided by the network driver for
-	// node-local networks. Only thing needed is populating the driver's name
-	// in the driver's state.
+	// node-local networks. Populate the driver's name, options and also
+	// the IPAM config options, all of these are needed to create
+	// the networks on nodes.
 	if nw.isNodeLocal {
 		n.DriverState = &api.Driver{
 			Name: d.name,
 		}
+
+		if n.Spec.DriverConfig != nil {
+			n.DriverState.Options = n.Spec.DriverConfig.Options
+		}
+
 		// In order to support backward compatibility with older daemon
 		// versions which assumes the network attachment to contains
 		// non nil IPAM attribute, passing an empty object
 		n.IPAM = &api.IPAMOptions{Driver: &api.Driver{}}
+
+		if n.Spec.IPAM != nil {
+			if n.Spec.IPAM.Driver != nil {
+				n.IPAM.Driver.Name = n.Spec.IPAM.Driver.Name
+				n.IPAM.Driver.Options = n.Spec.IPAM.Driver.Options
```

Comment:
Please follow ```allocateDriverState``` to set the options correctly.

Reply:
Ah I see, you mean the combination of the probably received operational state from the store with the network spec.